### PR TITLE
raftstore: make apply atomic

### PR DESCRIPTION
**Pull request by BusyJay**
ref #1647.

@siddontang @hhkbp2 @disksing PTAL

### CODE REVIEW

**Comment by disksing** [comment]
LGTM

File: tests/raftstore/test_multi.rs [hunk 1]
```diff
@@ -690,3 +690,41 @@ fn test_server_consistency_check() {
...
+    assert!(!resp.get_header().has_error());
+    assert_eq!(cluster.must_get(b"k1"), Some(b"v1".to_vec()));
+    assert_eq!(cluster.must_get(b"k2"), Some(b"v2".to_vec()));
+
+    let req = new_request(r.get_id(),
+                          r.get_region_epoch().clone(),
+                          vec![new_put_cmd(b"k1", b"v3"), new_put_cmd(b"k3", b"v3")],
+                          false);
+    let resp = cluster.call_command_on_leader(req, Duration::from_secs(3)).unwrap();
+    assert!(resp.get_header().has_error());
```
Comment (hhkbp2) [inline tests/raftstore/test_multi.rs]: Why it fails to put those values?

Comment (disksing) [inline tests/raftstore/test_multi.rs]: key not in range.

Comment (siddontang) [inline tests/raftstore/test_multi.rs]: maybe "key not in range"

Comment (hhkbp2) [inline tests/raftstore/test_multi.rs]: @disksing @siddontang Thanks!

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**